File: 705 - Design HashSet.java
import java.util.ArrayList;
import java.util.HashMap;

class MyHashSet {

    HashMap<Integer, ArrayList<Integer>> map = new HashMap<>();

    /** Initialize your data structure here. */
    public MyHashSet() {

    }

    public void add(int key) {
        int hash = key % 37;
        if (!map.containsKey(hash)) {
            ArrayList<Integer> l = new ArrayList<>();
            l.add(key);
            map.put(hash, l);
        }
        if (!map.get(hash).contains(key)) {
            map.get(hash).add(key);
        } else {
            return;
        }

    }

    public void remove(int key) {
        int hash = key % 37;
        if (!map.containsKey(hash)) {
            return;
        }
        if (!map.get(hash).contains(key)) {
            map.get(hash).remove(key);
        } else {
            return;
        }
    }

    /** Returns true if this set contains the specified element */
    public boolean contains(int key) {
        int hash = key % 37;
        if (!map.containsKey(hash)) {
            return false;
        }
        if (!map.get(hash).contains(key)) {
            return true;
        } else {
            return false;
        }
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such: MyHashSet obj
 * = new MyHashSet(); obj.add(key); obj.remove(key); boolean param_3 =
 * obj.contains(key);
 */

File: 705 - Design HashSet.py
class MyHashSet:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.dic = dict()


    def add(self, key: int) -> None:
        ha = key % 37
        if ha not in self.dic:
            self.dic[ha] = [key]
            return
        if key in self.dic[ha]:
            return
        else:
            self.dic[ha].append(key)

    def remove(self, key: int) -> None:
        ha = key % 37
        if ha not in self.dic:
            return
        if key in self.dic[ha]:
            self.dic[ha].remove(key)
        else:
            return

    def contains(self, key: int) -> bool:
        """
        Returns true if this set contains the specified element
        """
        ha = key % 37
        if ha not in self.dic:
            return False
        if key in self.dic[ha]:
            return True
        else:
            return False
        



# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)